src/System.Net.Requests/src/System/Net/FtpWebRequest.cs

```diff
@@ -0,0 +1,2544 @@
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.AppendFile,
+                              FtpOperation.AppendFile,
+                              FtpMethodFlags.IsUpload
+                              | FtpMethodFlags.TakesParameter,
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.DeleteFile,
+                              FtpOperation.DeleteFile,
+                              FtpMethodFlags.TakesParameter,
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.GetDateTimestamp,
+                              FtpOperation.GetDateTimestamp,
+                              FtpMethodFlags.TakesParameter,
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.GetFileSize,
+                              FtpOperation.GetFileSize,
+                              FtpMethodFlags.TakesParameter,
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.Rename,
+                              FtpOperation.Rename,
+                              FtpMethodFlags.TakesParameter,
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.MakeDirectory,
+                              FtpOperation.MakeDirectory,
+                              FtpMethodFlags.TakesParameter
+                              | FtpMethodFlags.ParameterIsDirectory,
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.RemoveDirectory,
+                              FtpOperation.RemoveDirectory,
+                              FtpMethodFlags.TakesParameter
+                              | FtpMethodFlags.ParameterIsDirectory,
+                              null),
+            new FtpMethodInfo(WebRequestMethods.Ftp.PrintWorkingDirectory,
+                              FtpOperation.PrintWorkingDirectory,
+                              FtpMethodFlags.DoesNotTakeParameter,
+                              null)
+        };
+
+    }
+
+    /// <summary>
+    /// <para>The FtpWebRequest class implements a basic FTP client
+    /// interface.</para>
+    /// </summary>
+    public sealed class FtpWebRequest : WebRequest {
+        private object          m_SyncObject;
+        private ICredentials    m_AuthInfo;
+        private readonly Uri    m_Uri;
+        private FtpMethodInfo   m_MethodInfo;
+        private string          m_RenameTo = null;
+        private bool            m_GetRequestStreamStarted;
+        private bool            m_GetResponseStarted;
+        private DateTime        m_StartTime;
+        private int             m_Timeout = s_DefaultTimeout;
+        private int             m_RemainingTimeout;
+        private long            m_ContentLength = 0;
+        private long            m_ContentOffset = 0;
+        private IWebProxy       m_Proxy;
+#if !FEATURE_PAL
```

Comment:
CodeFormatter will help with this, but all of the member variables should be renamed according to corefx guidelines as well, e.g. m_Proxy => _proxy, m_StartTime => _startTime, etc.